jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/quicknewjob/NewJobPresenterTest.java

```diff
@@ -0,0 +1,239 @@
+                                                      anyString(),
+                                                      any(Date.class),
+                                                      any(Map.class));
+
+        presenter.createJob(JOB_NAME,
+                            new Date(),
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            new ArrayList<RequestParameterSummary>());
+
+        verify(executorServicesMock).scheduleRequest(anyString(),
+                                                     anyString(),
+                                                     any(Date.class),
+                                                     any(HashMap.class));
+    }
+
+    @Test
+    public void InvalidClass_shouldCauseValidationError() {
+        doAnswer(new Answer() {
+            @Override
+            public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
+                throw (new Exception("Invalid command type"));
+            }
+        }).when(executorServicesMock).scheduleRequest(anyString(),
+                                                      anyString(),
+                                                      any(Date.class),
+                                                      any(Map.class));
+
+        presenter.createJob(JOB_NAME,
+                            new Date(),
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            new ArrayList());
+
+        verify(view).cleanErrorMessages();
+        verify(view).showInvalidTypeErrorMessage();
+    }
+
+    @Test
+    public void genericSchedulingErrorTest() {
+        String genericErrorMessage = "Unexpected error";
+        doAnswer(new Answer() {
+            @Override
+            public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
+                throw (new Exception(genericErrorMessage));
+            }
+        }).when(executorServicesMock).scheduleRequest(anyString(),
+                                                      anyString(),
+                                                      any(Date.class),
+                                                      any(Map.class));
+
+        presenter.createJob(JOB_NAME,
+                            new Date(),
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            new ArrayList());
+
+        verify(view).cleanErrorMessages();
+        verify(view).showInlineNotification(eq(genericErrorMessage));
+    }
+
+    @Test
+    public void invalidSchedulingError_EmptyNameTest() {
+        presenter.createJob("",
+                            new Date(),
+                            JOB_TYPE,
+                            JOB_RETRIES,
+                            new ArrayList());
+
+        verify(view).cleanErrorMessages();
+        verify(view).showEmptyNameErrorMessage();
```

Comment:
Maybe it makes sense to also add the verification that _ExecutorService_ never called _scheduleRequest_.

Reply:
+1